# Caso BOPS

In [1]:
import pandas as pd
import numpy as np

Primero comparamos las ventas totales que se generaron antes y después del lanzamiento de la iniciativa para ver si hubo ganancias o pérdidas.

In [2]:
bops = pd.read_csv('/home/lorena/Escritorio/OPI/BOPS_case/BOPS_case/bops_bm.csv', encoding='utf-8')

El archivo tiene varios huecos, así que hay que removerlos:

In [3]:
for i in range(7,11):
    del(bops['Unnamed: '+str(i)])

In [4]:
bops.dropna(inplace=True)

En la columna **sales** python a sus elementos como cadenas, además, tienen un coma que es necesario quitar si se quieren convertir a enter, así que a continuación se eliminará ese caracter y se convertirán los elementos a enteros:

In [5]:
bops[' sales '] = [int(x.replace(',','')) for x in bops[' sales ']]

Hay que usar la columna **after** para separar las ventas antes y después del lanzamiento:

In [6]:
before_bops = bops[bops['after']==0]

In [7]:
after_bops = bops[bops['after']==1]

Ahora sólo hay que sacer la diferencia del total de ventas de ambos casos:

In [8]:
after_bops[' sales '].sum()-before_bops[' sales '].sum()

-17111578

Se observa que las ventas totales después de BOPS son menores que las ventas de antes de la iniciativa, por lo que hay pérdidas.

Para saber qué es lo que está pasando con las ventas en EUA y canadá, también separaré de acuerdo al país.

In [9]:
before_bops_usa = before_bops[before_bops['usa']==1]

In [10]:
before_bops_canada = before_bops[before_bops['usa']==0]

In [11]:
after_bops_usa = after_bops[after_bops['usa']==1]

In [12]:
after_bops_canada = after_bops[after_bops['usa']==0]

In [13]:
loss_usa = after_bops_usa[' sales '].sum()-before_bops_usa[' sales '].sum()

In [14]:
loss_canada = after_bops_canada[' sales '].sum()-before_bops_canada[' sales '].sum()

In [15]:
loss_usa

-12275086

In [16]:
loss_canada

-4836492

In [17]:
loss_usa/loss_canada*100

253.80143293941146

Las pérdidas de EUA son de más de 12 millones, mientras que las de Canadá son de las de 4 millones, de hecho, las pérdidas de EUA son casi 254% mayores a las de Canadá. Así que me parece factible una expansión a Canadá, ya que no se vislumbran tantas pérdidas como en el otro país; en el momento que se hizo el análisis, BOPS tenía muy pocos meses en marcha, y se sabe que el sistema es complejo, sólo es necesario capacitar a los trabajadores al nuevo sistema y en poco tiempo comenzarán a recuperarse.